lib/chef/data_collector/messages/helpers.rb

```diff
@@ -0,0 +1,161 @@
+#
+# Author:: Adam Leff (<adamleff@chef.io)
+# Author:: Ryan Cragun (<ryan@chef.io>)
+#
+# Copyright:: Copyright 2012-2016, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class Chef
+  class DataCollector
+    module Messages
+      module Helpers
+        #
+        # Fully-qualified domain name of the Chef Server configured in Chef::Config
+        # If the chef_server_url cannot be parsed as a URI, the node["fqdn"] attribute
+        # will be returned, or "localhost" if the run_status is unavailable to us.
+        #
+        # @param run_status [Chef::RunStatus] The RunStatus object for this Chef Run.
+        #
+        # @return [String] FQDN of the configured Chef Server, or node/localhost if not found.
+        #
+        def chef_server_fqdn(run_status)
+          if !Chef::Config[:chef_server_url].nil?
+            URI(Chef::Config[:chef_server_url]).host
+          elsif run_status
+            run_status.node["fqdn"]
```

Comment:
In actuality, we're not using this value to connect to anything... it's simply to report what chef server (if in client mode) we used... otherwise, what host are we on. `node_name` may actually do the trick here.